aws-android-sdk-ddb-mapper/src/main/java/com/amazonaws/mobileconnectors/dynamodbv2/dynamodbmapper/S3ClientCache.java

```diff
@@ -109,39 +74,9 @@ public AmazonS3Client getClient(Region s3region) {
         if (client != null) {
             return client;
         }
-        if (credentialProvider != null) {
-            client = new AmazonS3Client(credentialProvider);
-        } else {
-            client = new AmazonS3Client(credentials);
-        }
+        client = new AmazonS3Client(credentialProvider, s3region.toAWSRegion());
         client.setRegion(s3region.toAWSRegion());
         final AmazonS3Client prev = clientsByRegion.putIfAbsent(s3region, client);
         return prev == null ? client : prev;
     }
-
-    /**
-     * Returns a {@link TransferManager} for the given region, or throws an
-     * exception when unable. The returned {@link TransferManager} will always
-     * be instantiated from whatever {@link AmazonS3Client} is in the cache,
-     * whether provided with {@link #useClient(AmazonS3Client)} or instantiated
-     * automatically from {@link AWSCredentials}. Any {@link TransferManager}
-     * returned could be shut down if a new underlying {@link AmazonS3Client} is
-     * provided with {@link #useClient(AmazonS3Client)}.
-     *
-     * @param region The region the returned {@link TransferManager} will be
-     *            configured to use.
-     * @return A transfer manager for the given region from the cache, or one
-     *         instantiated automatically from any existing
-     *         {@link AmazonS3Client},
-     */
-    public TransferManager getTransferManager(Region region) {
```

Comment:
TLDR, it wasn't actually being used by the mapper? Or why is this safe?

Reply:
It was not being used outside of integration tests. Since this method directly returns a deprecated item, I think it is within scope to remove this too.

The main purpose of this class was to map S3 clients by region. It also associated each of those clients to a transfer manager instance, but this was just to make it easier for customers to use those S3 clients. Nowhere in the codebase was the transfer manger being explicitly used though.